### PR TITLE
Slight amendment to requirement 10

### PR DESCRIPTION
**Pull request by vapourer**
Suggestion for a slight amendment to requirement 10.  Hope it might be a little more specific, and that the rationale points to the project purpose

### CODE REVIEW

**Comment by MrfksIv** [comment]
I am not so sure about the usage of the word 'predict' as it could imply something quite more complicated than just subtracting the needed resources from the stocked ones.

Also what are 'regularly used' resources? Does this mean it will notify staff only for some resources and not for others?

I believe that the problem with 010  is that it implicitly contains more than one requirements.
Maybe it should be broken down into two or possibly three separate requirements?

"The product shall keep track of additional equipment requirements and stocked equipment"
"The product shall  estimate deficits in equipment based on current stock levels and total additional requirements of scheduled workshops"
"The product shall notify the staff of any deficits in equipment"

Maybe I would merge the second and third requirement into one however one should consider where the product boundary lies. By keeping it as a separate requirement, one could consider automating the ordering process instead of just notifying staff.

**Comment by vapourer** [comment]
I have always guessed that requirement 10 was based on the discussion with Karen from the TMA02 transcript.  I suggested the word 'predict' because of Karen's phrase 'several of the facilitators have worked with us for years so we have a fair idea of what they will need'.  I can envisage an automated system capable of providing that functionality.
On the other hand, I cannot imagine a system that can 'identify additional equipment requirements'.  My quibble is with the word 'additional'.  If the latter is the correct interpretation, I think the requirement should read something like 'The product shall accept additional equipment requirements', clarifying the element of user input, ie an external business event.
I know what you mean about splitting the requirement, and considered it, but baulked in the end, for pragmatic reasons as much as anything.  We are not going to be able to complete a full specification.  It seems to me that the target here is to agree a set of functional requirements so that we all have one to work with, and from which we are each confident we can derive one non-functional requirement.  Then we have to be in a position to review each other's derived non-functional requirements.  My reading of the quality gateway chapter suggests that the review process will include tracing back to the rationale in the associated functional requirement, and that that is a sub-element of the question.  Might be wrong :-).  If this were for real, I suspect we would indeed be looking at creating at least two requirements from this one, probably along the lines you have indicated, but I would still question some uses of the word 'additional'.  As it is, I have guessed that the group size is 17 (the original requirements list reached 34), so we need 17 usable requirements.  While I agree with you in principle, unless we really feel compelled, I wonder whether we should avoid further splitting requirements at this stage.
Andrew, what are your feelings?  This was your requirement, and certainly seems important.  Is the requirement more about monitoring, or is it about recording, the latter involving user input?

**Comment by tim-hm** [comment]
I think we're in danger of spending too much time on debating wording when we should focus reaching an agreed set of requirements to enable the next assignment questions.

There's an artificial element in the way we're defining the requirements. In the real world, we would be in a team with set roles and therefore someone would have the authority to make a final decision.  Although I am sure we would eventually arrive at a consensus it would likely be after the assignment is due!

File: WaltonSustainabilityCentre-Requirements.txt
Comment (AdrianPotter) [inline WaltonSustainabilityCentre-Requirements.txt:51]: I'm not keen on the wording of this one. I know we don't have fit criteria yet but:

1. This suggests two separate atomic requirements
    1. Predicting stock levels
    1. Providing notifications
1. The interpretation of "regularly used" is ambiguous and could vary greatly.

**Review by AdrianPotter — DISMISSED** [review]

**Comment by vapourer** [comment]
'I am sure we would eventually arrive at a consensus it would likely be after the assignment is due'
That sounds a tad optimistic to me :-)

**Comment by AdrianPotter** [comment]
I rushed a bit headlong into this one, but the review I originally submitted rather agrees with the original assessment. My interpretation of the original requirements is that the system would do some sort of operation like:

```
final_resources = current_resources - requested_resources
if ( final_resource < minimum_resource_level ) {
    notify()
}
```

Pseudo-code for illustration purposes only. It may be that phrasing the requirement as:

```
The product shall notify users when planned resource usage will drop stocks below their minimum level
```

That contains all of the functionality you suggested before but as a single rather than multiple functions. At the end of the day, it depends how atomic we want to be.

In terms of process we have two choices:

1. Merge (with or without edits)
2. Don't merge

There's nothing wrong with not accepting a change if we don't want to include it. If we can't reach a consensus then we probably shouldn't merge. There are many more requirements than users so there's minimal risk of someone getting stuck with an unworkable requirement for TMA03.

**Comment by vapourer** [comment]
One thing that's been at the back of my mind throughout the weekend ...
... I have no intention of applying for a business analyst's job any time soon
:-)

**Comment by vapourer** [comment]
In terms of process we have two choices:
Merge (with or without edits)
Don't merge
There's nothing wrong with not accepting a change if we don't want to include it. If we can't reach a consensus then we probably shouldn't merge. There are many more requirements than users so there's minimal risk of someone getting stuck with an unworkable requirement for TMA03.

That is true, but as there's no voting system, the lack of responses means there's no apparent decision.  Otherwise it would be easy to withdraw the pull request.  And I do agree with Timothy about the artificiality of this exercise.  Ultimately, all we need is a functional requirement each.  My reading of the set book has suggested that wordings of both descriptions and rationales do matter, but if we can't agree in this forum, that is soon dealt with on an individual basis in the TMA.  I think your setting this up was great, but I also think the activity needed far greater participation and turnover of pull requests than has been apparent.  Also, as I suggested in a forum post, rather than arguing too much over details in each pull request, I think an iterative incremental approach would have worked very well ie had enough of us engaged, if a pull request has elements that are useful, but others that you don't like, merge it anyway and submit another pull request to correct the part you object to.  There has not been sufficient momentum to work like that :-(

File: WaltonSustainabilityCentre-Requirements.txt
Comment (radvansky-tomas) [reply WaltonSustainabilityCentre-Requirements.txt:51]: I think we should have "EPIC" tasks like high level overview of minimum set of requirements to fulfil idea of project. What you are suggesting is specifying this requirement more, which in fact does not help at all. 

I suggest to have generic tasks like:
- Track equipment for events

Which as you suggested should contain in future subtasks of prediction, notifications etc..

**Comment by AdrianPotter** [comment]
It's hard to imagine a group exercise that doesn't require group engagement unfortunately. I agree that there may be some element of individual tweaking of requirements as part of TMA03 Q2.

In terms of iterative approaches, are you aware that if you submit more commits on a branch that is associated with the pull request, the change will automatically become part of the request? You can see that in my PR: https://github.com/Open-University-PG-SWEng/M814-16K-Tyrrell/pull/6. In that sense, each pull request should be an iterative process in its own right.

**Comment by vapourer** [comment]
I was creating a discrete branch for each pull request.  My experience of git is limited, but, during the three months I worked with it, we had a new branch per work item, task, bug or whatever.  I understand that branches are _cheap_ in git
From the look of your pull requests, I had guessed that you were doing that too.  But, as I say, I'm not that confident with git

**Comment by vapourer** [comment]
But yes, I did know that, and amended one of my own pull requests accordingly.  I didn't touch anybody else's like that

**Pull request closed without merging**
